### ISSUE TITLE
V2.0.8, Enabling Decoy Password on the Password Pump freezes the Pump

### ISSUE THREAD

**Issue by scootercommuter** [seawarrior181/PasswordPump_II#11]
On the Password Pump itself, Actions:Settings, Decoy Password locks up the Password Pump.  Have to hit the Reset button to regain control.  Show Password, RGB LED, Timeout Minutes, Login Attempts all work, I didn't try the other settings.  Choosing Decoy Password in the GUI seems to work, but I haven't tested whether it actually erases stuff.  Now I see that this freezeup is ONLY applicable to the Joystick version.

**Comment by seawarrior181** [comment]
Hi ScooterCommuter,

Thank you for reporting this defect  to me.  Are you able to reproduce the behavior at will?  I am not able to reproduce this error on my side.  What I have noticed is that for three settings, Decoy Password, Keyboard (not Keyboard Language), and Show Password,  after the device has been factory reset and before the user has set any of these settings, when first entering the menu it doesn't indicate if the setting is On or Off; it defaults to blank.  If you select blank it stays blank, but it doesn't freeze.  And selecting On or Off doesn't freeze it either, at least on my end.  Would you be able to try to see if you can make the PasswordPump freeze on demand?  Frankly I'd prefer it if we could get it to freeze consistently because intermittent defects can be extremely difficult to isolate and fix.  Thanks again for the report.  Parenthetically, I am working on a new version of the firmware this morning which addresses another defect I noticed on the joystick model when changing orientation.  I'll get to your other report in a bit.  Also, please let me know what version of the firmware your running, the date that displays when you power on your device, and the color of the PCB.  Thanks!

Best,
Dan

**Comment by seawarrior181** [comment]
Ok I have the version information, no need to re-report that.

**Comment by seawarrior181** [comment]
ScooterCommuter, when you say the freezeup is only applicable to the joystick model, do you mean that you're not able to reproduce the problem on the rotary encoder model but you are able to reproduce the issue on the joystick model?  Sadly I am not able to reproduce the problem on either model w/ firmware Version:2.0.8.02.

Thanks,

Dan

**Comment by scootercommuter** [comment]
Yes, correct.  Always happens with my Joystick M0 model.  Not reproducible on Joystick M4, not reproducible on Rotary Encoder M0.

**Comment by seawarrior181** [comment]
ScooterCommuter,

There's no behavior surrounding the management of that setting (decoy password on/off) that 'cares' about the model or the MCU type, and I'm not able to reproduce the behavior on my  M0 joystick model.  My only suggestion would be to upgrade to the latest version of the firmware and try again.  Please let me know if that addresses the issue.  If the issue persists, please answer the following question:  When you go to Settings->Decoy Password, does it display On, Off, or blank?

Best,
Dan
